packages/web/src/store/application/ui/buy-audio/sagas.ts

```diff
@@ -1080,14 +1081,19 @@ function* watchStartBuyAudioFlow() {
 }
 
 function* watchRecovery() {
-  yield* call(recoverPurchaseIfNecessary)
+  yield takeLeading(startRecoveryIfNecessary, recoverPurchaseIfNecessary)
```

Comment:
Maybe a comment here indicating why we're using takeLeading, this is gonna be very mysterious 6mo from now for subsequent maintainers